### PR TITLE
So/apply filter request

### PR DESCRIPTION
**Pull request by spanersoraferty**

### CODE REVIEW

File: Dfe.Data.SearchPrototype/Web/Controllers/HomeController.cs [hunk 1]
```diff
@@ -27,18 +32,30 @@ public class HomeController : Controller
...
+    /// </param>
+    /// <param name="requestMapper">
+    /// The concrete implementation of the T:DfE.Data.ComponentLibrary.CrossCuttingConcerns.Mapping.IMapper<Dictionary<string, List<string>>, IList<FilterRequest>>
+    /// defined within, and injected by the IOC container (defined within program.cs) used to map the user input to a list of <see cref="FilterRequest"/> request types.
+    /// </param>
     public HomeController(
         ILogger<HomeController> logger,
         IUseCase<SearchByKeywordRequest, SearchByKeywordResponse> searchByKeywordUseCase,
-        IMapper<SearchByKeywordResponse, SearchResultsViewModel> mapper)
+        IMapper<EstablishmentResults?, List<ViewModels.Establishment>?> establishmentResultsToEstablishmentsViewModelMapper,
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/Web/Controllers/HomeController.cs:50]: We have quite a mix of IList and List going on - I did changes on the train yesterday to make as much as I could see into IList - maybe best done as a separate PR after this one though

File: Dfe.Data.SearchPrototype/Web/Controllers/HomeController.cs [hunk 2]
```diff
@@ -54,13 +71,56 @@ public async Task<IActionResult> Index(string searchKeyWord)
...
+                        searchKeyword: searchRequestViewModel.SearchKeyword + "*",
+                        filterRequests: _requestMapper.MapFrom(searchRequestViewModel.SelectedFacets!)));
+
+            ViewModels.SearchResults viewModel = new()
+            {
+
+                SearchItems = _establishmentResultsToEstablishmentsViewModelMapper.MapFrom(response.EstablishmentResults),
+                Facets = _establishmentFacetsToFacetsViewModelMapper.MapFrom(new EstablishmentFacetsMapperRequest(
+                        response.EstablishmentFacetResults, searchRequestViewModel.SelectedFacets))
+            };
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/Web/Controllers/HomeController.cs:122]: I took the CreateViewModel private method out - was thinking maybe it points to us needing to do it somewhere else

File: Dfe.Data.SearchPrototype/Web/Mappers/EstablishmentFacetsToFacetsViewModelMapper.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="input">
+        /// 
+        /// </param>
+        /// <returns>
+        /// 
+        /// </returns>
+        public List<Facet>? MapFrom(EstablishmentFacetsMapperRequest input)
```
Comment (CathLass) [inline Dfe.Data.SearchPrototype/Web/Mappers/EstablishmentFacetsToFacetsViewModelMapper.cs:21]: I don't really like this EstablishmentFacetsMapperRequest - maybe it's just the name I don't like - or could we just send a Tuple 😱

Comment (CathLass) [reply Dfe.Data.SearchPrototype/Web/Mappers/EstablishmentFacetsToFacetsViewModelMapper.cs:21]: rename instead of request

**Comment by CathLass** [comment]
In a separate PR we will rename models as ViewModels as appropriate, all inside Models folder (probably)

**Comment by CathLass** [comment]
The unit tests on the controller don't test that the mappers are called with the correct params, this will be fixed when we refactor to use a factory method for creating the view model

**Comment by CathLass** [comment]
Some comments are missing - we cheated by putting in empty ones :)

File: Dfe.Data.SearchPrototype/Web/Mappers/FacetsAndSelectedFacetsToFacetsViewModelMapper.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
+﻿using Dfe.Data.SearchPrototype.Common.Mappers;
+using Dfe.Data.SearchPrototype.SearchForEstablishments.Models;
+using Dfe.Data.SearchPrototype.Web.Models;
+
+namespace Dfe.Data.SearchPrototype.Web.Mappers
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public sealed class FacetsAndSelectedFacetsToFacetsViewModelMapper : IMapper<FacetsAndSelectedFacets, List<Facet>?>
```
Comment (AsiaWi) [inline Dfe.Data.SearchPrototype/Web/Mappers/FacetsAndSelectedFacetsToFacetsViewModelMapper.cs:10]: It would make more sense to rename models in mid tier to match the view model. Example: EstablishmentFacet to Facet and FacetResult to FacetValue

**Review by CathLass — COMMENTED** [review]
We need to rename variables and test doubles in line with the class renames

File: Dfe.Data.SearchPrototype/Web/Mappers/EstablishmentFacetsToFacetsViewModelMapper.cs (hunk 3 above)
Comment (spanersoraferty) [reply Dfe.Data.SearchPrototype/Web/Mappers/EstablishmentFacetsToFacetsViewModelMapper.cs]: Rename completed.

File: Dfe.Data.SearchPrototype/Web/Mappers/FacetsAndSelectedFacetsToFacetsViewModelMapper.cs (hunk 4 above)
Comment (CathLass) [reply Dfe.Data.SearchPrototype/Web/Mappers/FacetsAndSelectedFacetsToFacetsViewModelMapper.cs:12]: Thanks for offering to do that Asia :)

**Review by CathLass — APPROVED** [review]
stories added to board for work to happen offline.
